cmd/main.go

```diff
@@ -1,17 +1,35 @@
 package main
 
 import (
-	"log"
 	"net/http"
 
+	"github.com/gin-gonic/gin"
 	"github.com/tech-pool/takemetotheshop-api/api/registration"
 )
 
 const address string = ":8080"
 
 func main() {
 
-	http.HandleFunc("/registration", registration.Handler)
+	router := setupRouter()
+	router.Run(address)
+
+	//http.HandleFunc("/registration", registration.Handler)
+	// log.Fatalln(http.ListenAndServe(address, nil))
+}
+
+func setupRouter() *gin.Engine {
+
+	///Engine  returned by Default() is the framework's instance,
+	//it contains the muxer, middleware and configuration settings.
+	r := gin.Default()
+
+	r.GET("/", func(c *gin.Context) {
+		c.String(http.StatusOK, "Hello Gin")
+	})
+
+	r.GET("/registration", registration.Handler)
```

Comment:
Should we have a /api prefix to all our API routes? More as a good convention? And maybe make api a package and all the routes part of the api package?

Reply:
was advised each functionality to have its own package

Reply:
Not denying that, but won't it be easier if the starting main file is slim and the APIs are in a single package? All functionalities can be individual packages inside of that API package?